### PR TITLE
Add support for email verification

### PR DESCRIPTION
**Pull request by mixonic**
@bantic for you to review/merge. /cc @fancyremarker we worked on this yesterday.

Adds support to diesel for email verification. This is still a stand-alone feature, and not part of the wizard flow.

Also adds a top level error page, required to show a default message for invalid tokens.

### CODE REVIEW

File: app/error/controller.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export default Ember.Controller.extend({
+
+  title: function(){
+    return this.get('model.status') || this.get('model.name');
+  }.property('model.status', 'model.name'),
+
+  subTitle: function(){
+    return this.get('model.statusText');
+  }.property('model.statusText', 'model.message'),
```
Comment (bantic) [inline app/error/controller.js]: is it actually dependent on model.message?

File: app/models/verification.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import DS from 'ember-data';
+
+export default DS.Model.extend({
+  verification_code: DS.attr('string'),
```
Comment (bantic) [inline app/models/verification.js]: Is there a reason for not having this camelCased `verificationCode`?

File: app/utils/url-parameters.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export function getUrlParameters(loc) {
+  var a = loc.search.substr(1).split('&');
+  var b = {};
+  if (a === "") {
```
Comment (bantic) [inline app/utils/url-parameters.js]: wouldn't `a` be an array? And if there's no search it should be `[""]` then?

**Comment by bantic** [comment]
@mixonic sweet — I added a few qs there, after you take a look at those go ahead and merge at your leisure

**Comment by mixonic** [comment]
@bantic updated to address your comments, and added unit tests for the url params parser. Thanks!

**Pull request merged**